bin/index.js

```diff
@@ -0,0 +1,666 @@
+      ifError(err)
+    }).on('end', function () {
+      each(stamps, function (stamp, cb) {
+        clocker.set(stamp, 'archive', value, cb)
+      }, nil)
+    }).on('data', function (entry) {
+      stamps.push(entry.key)
+    })
+  } else {
+    clocker.set(stamp, 'archive', value, nil)
+  }
+}
+
+function archive (stamp, cmd) {
+  setArchive(true, stamp, cmd)
+}
+
+function unarchive (stamp, cmd) {
+  setArchive(false, stamp, cmd)
+}
+
+function edit (stamp, prop, cmd) {
+  if (cmd) {
+    clocker = initialize(cmd)
+  }
+
+  clocker.get(stamp, function (err, obj) {
+    if (err) {
+      if (err.name === 'NotFoundError' && typeof stamp !== 'undefined' && typeof prop === 'undefined') {
+        // stamp was just prop
+        return edit(null, stamp, null)
+      } else {
+        ifError(err)
+      }
+    }
+
+    obj.data = obj.data || {}
+    if (obj.end && obj.end !== 'NOW') {
+      obj.data.end = strftime('%F %T', obj.end)
+    }
+
+    var src = obj.data
+    if (typeof prop !== 'undefined') {
+      if (obj.data.hasOwnProperty(prop)) {
+        src = obj.data[prop]
+      } else {
+        ifError(new Error('Property not set: ' + prop))
+      }
+    }
+
+    src = stringify(src, {
+      space: 2
+    })
+    editEntry(src, function (updated) {
+      try {
+        updated = JSON.parse(updated)
+      } catch (err) {
+        ifError(new Error('error parsing json'))
+      }
+
+      if (typeof prop !== 'undefined') {
+        clocker.set(stamp, prop, updated, nil)
+      } else {
+        clocker.set(stamp, updated, nil)
+      }
+    })
+  })
+}
+
+function initialize (cmd) {
+  if (typeof cmd !== 'object') {
+    program.outputHelp()
+    process.exit(1)
+  }
+  return new Clocker({
+    dir: dir(cmd)
+  })
+}
+
+function getFilter (cmd) {
+  var filter = {}
+  filter.test = (e) => true
+
+  if (cmd.type) {
+    if (cmd.type[0] === '/' && cmd.type.slice(-1)[0] === '/') {
+      // regex
+      filter.test = (e) => RegExp(cmd.type.slice(1, -1)).test(e.data.type)
+    } else {
+      // string
+      filter.test = (e) => e.data.type === cmd.type
+    }
+  }
+
+  if (cmd.gt) {
+    filter.gt = cmd.gt
+  }
+  if (cmd.lt) {
+    filter.lt = cmd.lt
+  }
+
+  var oldTest = filter.test
+  if (!cmd.all) {
+    // filter archived records
+    filter.test = (e) => {
+      if (!oldTest(e)) {
+        return false
+      }
+      if (e.data && e.data.archive) {
+        return false
+      }
+      return true
+    }
+  }
+
+  return filter
+}
+
+function dir (cmd) {
+  if (cmd.datadir) {
+    return cmd.datadir
+  }
+
+  var HOME = process.env.HOME || process.env.USERPROFILE
+  var defaultDataDir = path.join(HOME, '.clocker')
+  return defaultDataDir
+}
+
+function each (arr, cb, cbEnd) {
+  if (arr.length === 0) {
+    return cbEnd()
+  }
+
+  cb(arr.shift(), function (err) {
+    if (err) {
+      return cbEnd(err)
+    }
+    each(arr, cb, cbEnd)
+  })
+}
+
+function started (err, stamp) {
+  ifError(err)
+  success()
```

Comment:
You can call `nil()` instead, since it does the same.

Reply:
Replaced by `nil()`.